sdk/src/main/java/com/amplitude/experiment/DefaultExperimentClient.kt

```diff
@@ -263,13 +264,17 @@ internal class DefaultExperimentClient internal constructor(
             throw IOException("fetch error response: $response")
         }
         val body = response.body?.string() ?: ""
-        val json = JSONObject(body)
         val variants = mutableMapOf<String, Variant>()
-        json.keys().forEach { key ->
-            val variant = json.getJSONObject(key).toVariant()
-            if (variant != null) {
-                variants[key] = variant
+        try {
+            val json = JSONObject(body)
+            json.keys().forEach { key ->
+                val variant = json.getJSONObject(key).toVariant()
+                if (variant != null) {
+                    variants[key] = variant
+                }
             }
+        } catch (e: JSONException) {
```

Comment:
We should catch this exception from the place where parseResponse is called rather than inside this function. Line 238. Then we can call the `onFailure` function to complete the future exceptionally and propagate the error more clearly.